src/main/scala/org/apache/mesos/chronos/scheduler/config/MainModule.scala

```diff
@@ -102,12 +102,17 @@ class MainModule(val config: SchedulerConfiguration with HttpConf)
       curator = curator,
       leaderLatch = leaderLatch,
       leaderPath = config.zooKeeperCandidatePath,
-      notificationClients = notificationClients,
+      jobsObserver = jobsObserver,
       failureRetryDelay = config.failureRetryDelayMs(),
       disableAfterFailures = config.disableAfterFailures(),
-      jobMetrics = metrics,
-      jobStats = stats,
-      clusterName = config.clusterName.get)
+      jobMetrics = metrics)
+  }
+
```

Comment:
it is Guice @Provides annotated factory method, if I understand Guice correctly. So whenever JobsObserver instance is needed a singleton instance is provided. https://github.com/google/guice/wiki/ProvidesMethods